### PR TITLE
feature/delegated operators

### PR DESCRIPTION
**Pull request by nebulae**
## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested? If it is not, please explain why.

(Details)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

**Review by imanjra — COMMENTED** [review]
Nice work! 🔥 
Left some minor comment but overall looks good to me

File: fiftyone/factory/repo_factory.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+FiftyOne Server
```
Comment (benjaminpkane) [inline fiftyone/factory/repo_factory.py:2]: Suggestion
```suggestion
FiftyOne repository factory
```

File: fiftyone/factory/repos/delegated_operation_doc.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+from datetime import datetime
```
Comment (benjaminpkane) [inline fiftyone/factory/repos/delegated_operation_doc.py:1]: nit: missing module doc string

File: fiftyone/factory/repos/delegated_operation.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            operator=operator, dataset_id=dataset_id, run_state="queued"
+        )
+
+    def list_operations(
+        self,
+        operator: str = None,
+        dataset_id=None,
+        run_state: str = None,
+        delegation_target: str = None,
+        **kwargs: Any,
```
[the remark this answers is not in the document]
Comment (nebulae) [reply fiftyone/factory/repos/delegated_operation.py:147]: for teams, the user who initialed the run (likely in the context though) and potentially some permission info

**Pull request closed without merging**